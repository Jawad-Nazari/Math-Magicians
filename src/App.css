* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.quote-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 1.5rem;
  color: white;
  margin-bottom: 2rem;
}

.question {
  font-size: 1.5rem;
  color: white;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.quote {
  font-size: 1.5rem;
  color: white;
  width: 70%;
}
