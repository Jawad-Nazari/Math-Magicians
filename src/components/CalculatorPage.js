import React from 'react';

import Calculator from './calculator';
import Card from './UI/Card';
import classes from './CalculatorPage.module.css';

const CalculatorPage = () => (
  <section className={classes.calculator_section}>
    <Card extraclass={classes.calculatorPage_details}>
      <h3>Lets do some Math!</h3>
      <p>
        Using Our nice calculator that can handle basic arithmetic operations
      </p>
    </Card>
    <Calculator />
  </section>
);

export default CalculatorPage;
