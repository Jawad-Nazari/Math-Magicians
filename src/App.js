/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { Route, Routes } from 'react-router-dom';
import Layout from './components/UI/Layout';
import Quote from './components/quote';
import Home from './components/home';
import CalculatorPage from './components/CalculatorPage';

const App = () => (
  <div className="App">
    <header className="App-header">
      <Layout>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<CalculatorPage />} />
          <Route path="/quote" element={<Quote />} />
        </Routes>
      </Layout>

    </header>
  </div>
);

export default App;
