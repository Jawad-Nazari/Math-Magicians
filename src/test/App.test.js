/* eslint-disable import/no-extraneous-dependencies */
import { render, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../components/home';
import CalculatorPage from '../components/CalculatorPage';

import App from '../App';

describe('testing App.js with snapshot', () => {
  test('test App.js dom snapshot', () => {
    const app = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();

    expect(app).toMatchSnapshot();
  });

  describe('Home component', () => {
    it('renders the welcome message', () => {
      const { getByText } = render(<Home />);
      const headingElement = getByText(/Welcome to our page!/i);
      const paragraphElement = getByText(/Welcome to Math Magicians,/i);
      expect(headingElement).toBeInTheDocument();
      expect(paragraphElement).toBeInTheDocument();
    });
  });

  describe('CalculatorPage', () => {
    test('renders a card with a heading and paragraph', () => {
      const { getByText } = render(<CalculatorPage />);
      const headingElement = getByText('Lets do some Math!');
      const paragraphElement = getByText('Using Our nice calculator that can handle basic arithmetic operations');
      expect(headingElement).toBeInTheDocument();
      expect(paragraphElement).toBeInTheDocument();
    });
  });
  test('renders Quote component on Quote page', () => {
    render(
      <MemoryRouter initialEntries={['/quote']}>
        <App />
      </MemoryRouter>,
    );
    expect(screen.getByText('Did You Know?')).toBeInTheDocument();
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});
