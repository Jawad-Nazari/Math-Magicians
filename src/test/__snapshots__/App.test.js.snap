// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App.js with snapshot test App.js dom snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <header
      className="header"
    >
      <nav
        className="navbar"
      >
        <div
          className="brand"
        >
          <h1>
            <a
              href="/"
              onClick={[Function]}
            >
              Math Magicians
            </a>
          </h1>
        </div>
        <ul
          className="nav_items"
        >
          <li
            className="nav_item"
          >
            <a
              aria-current="page"
              className="nav_link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className="nav_link"
              href="/calculator"
              onClick={[Function]}
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              className="nav_link"
              href="/quote"
              onClick={[Function]}
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      className="main"
    >
      <div
        className="container"
      >
        <section
          className="home"
        >
          <h2>
            Welcome to our page!
          </h2>
          <p>
            Welcome to Math Magicians, your online destination for math enthusiasts! Our website features a powerful calculator that can handle basic arithmetic operations, Plus, our calculator is fully responsive, so you can use it on any device, from desktops to smartphones. In addition to our calculator, we also provide daily inspiration in the form of a random quote These quotes come from famous mathematicians and other notable figures in the world of science and technology, and are sure to motivate and inspire you as you tackle even the toughest math problems.
          </p>
        </section>
      </div>
    </main>
  </header>
</div>
`;
